distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
I'm fine rethinking things, though would prefer to do that thinking in a separate PR (and possibly much later).  I'd like to push on if possible.  Also happy for other people to take this on.

I'm curious about your thoughts on the explicit loop argument.  Can you explain more about how it is useless and fragile in our case?